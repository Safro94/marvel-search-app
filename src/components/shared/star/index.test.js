import React from 'react';
import { render, wait, waitForElement, fireEvent } from '@testing-library/react';

import Star from './';

jest.mock('../../icons/star', () => 'starIcon');
jest.mock('../../icons/fullStar', () => 'fullStarIcon');

jest.mock('./style', () => {
  return {
    Star: 'Star',
    FullStar: 'FullStar',
  };
});

describe('<Star /> ', () => {
  beforeEach(() => jest.clearAllMocks());
  const handleClick = jest.fn();

  it('It should render a FullStar if showFullStar is true', async () => {
    //Arrange
    const showFullStar = true;

    // Act
    const { container } = render(<Star showFullStar={showFullStar} handleClick={handleClick} />);
    await wait();

    // Assert
    expect(container).toMatchSnapshot();
  });

  it('It should render a Star if showFullStar is false', async () => {
    //Arrange
    const showFullStar = false;

    // Act
    const { container } = render(<Star showFullStar={showFullStar} handleClick={handleClick} />);
    await wait();

    // Assert
    expect(container).toMatchSnapshot();
  });

  it('When the fullStar is clicked, it should called the function passed by prop', async () => {
    //Arrange
    const showFullStar = true;

    // Act
    const { getByTestId } = render(<Star showFullStar={showFullStar} handleClick={handleClick} />);
    const fullStar = await waitForElement(() => getByTestId('full-star'));
    fireEvent.click(fullStar);

    // Assert
    expect(handleClick).toBeCalledTimes(1);
  });

  it('When the star button is clicked, it should called the function passed by prop', async () => {
    //Arrange
    const showFullStar = false;

    // Act
    const { getByTestId } = render(<Star showFullStar={showFullStar} handleClick={handleClick} />);
    const star = await waitForElement(() => getByTestId('star'));
    fireEvent.click(star);

    // Assert
    expect(handleClick).toBeCalledTimes(1);
  });
});
