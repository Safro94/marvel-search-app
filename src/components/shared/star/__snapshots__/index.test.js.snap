// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Star />  It should render a FullStar if showFullStar is true 1`] = `
<div>
  <fullstar
    data-testid="full-star"
  >
    <fullstaricon />
  </fullstar>
</div>
`;

exports[`<Star />  It should render a Star if showFullStar is false 1`] = `
<div>
  <star
    data-testid="star"
  >
    <staricon />
  </star>
</div>
`;
